custom_scripts/py_scripts/db_scanner/db_scanner.py

```diff
@@ -0,0 +1,255 @@
+            print(f"Host was unreachable: {ip_value} with {err}")
+            return None, err
+        if db_names is None:
+            return None
+        host_dict = {
+            "host": ip_value,
+            "port": port,
+            "version": client.server_info()["version"],
+        }
+        for db_num, db in enumerate(db_names):
+            try:
+                collection_names = client[db].list_collection_names()
+            except (AutoReconnect, OperationFailure, KeyError, InvalidName) as err:
+                print(f"Connection was over by server: {err}")
+                if err is AutoReconnect:
+                    if traceback.extract_tb(1) != ConnectionResetError:
+                        continue
+                return None, err
+            for col_num, collection_name in enumerate(collection_names):
+                # print(collection_name, " : ", col_num)
+                if len(collection_names) == 0:
+                    continue
+                try:
+                    host_dict[client[db][collection_name].name] = []
+                except InvalidName as err:
+                    print(f"Collection cannot be set: {err}")
+                    continue
+                try:
+                    try:
+                        docs_num = client[db][collection_name].count_documents({})
+                    except AutoReconnect as err:  # ConnectionResetError
+                        print(f"Connection was over by server: {err}")
+                        return None, err
+                    print(f"number of docs {docs_num} for {collection_name}")
+                    cursor = client[db][collection_name].find(limit=100)
+                    try:
+                        for document in cursor:
+                            document_list = find_values_by_key(document)
+                            if document_list is not None:
+                                host_dict[client[db][collection_name].name].append(
+                                    document_list
+                                )
+                    except InvalidBSON as err:
+                        print(f"Error occurred with decoding: {err}")
+                        if len(host_dict[client[db][collection_name].name]) == 0:
+                            host_dict.pop(client[db][collection_name].name)
+                        return host_dict, err
+                except (NetworkTimeout, CursorNotFound) as err:
+                    print(f"Waiting of server is too long: {err}")
+                    if len(host_dict[client[db][collection_name].name]) == 0:
+                        host_dict.pop(client[db][collection_name].name)
+                    save_results(host_dict)
+                    client.close()
+                    return host_dict, err
+                if len(host_dict[client[db][collection_name].name]) == 0:
+                    host_dict.pop(client[db][collection_name].name)
+        save_results(host_dict)
+        client.close()
```

Comment:
Use `with` statement, like:
```python3
with pymongo.MongoClient(...)
```
example:
https://programtalk.com/vs2/python/870/python-testdata/testdata/extra/mongodb.py/

In the case of `with` statement, you don't need to close the database connection, it will be closed automatically